src/components/ImageGrid/ImageGridComponent.tsx

```diff
@@ -98,29 +73,147 @@ const ImageGridComponent = ({
     } else {
       setRowCol({ col: 1, row: 1 });
     }
-  }, [numberOfImgs, images]);
-
-  // to avoid bug undefined url when user change number of images
-  useEffect(() => {
-    for (let i = 0; i < numberOfImgs; i++) {
-      images[i] = exampleImages[i];
-    }
   }, [numberOfImgs]);
+  // close Carousel when user press esc
+  useEffect(() => {
+    const keydownHandler = (evnt: KeyboardEvent) => {
+      evnt.key === "Escape" && setShowCarousel(false);
+    };
+    document.addEventListener("keydown", keydownHandler);
+    return () => document.removeEventListener("keydown", keydownHandler);
+  });
 
+  const useCarousel = (index: number) => {
+    showModal && setShowCarousel(true);
+    setSelectedImgIndex(index);
+  };
+  const closeCarousel = () => {
+    setShowCarousel(false);
+  };
   return (
-    <ImageGrid
-      height={imagesGridHeight}
-      width={imagesGridWidth}
-      numberOfImgs={numberOfImgs}
-      row={rowCol.row}
-      col={rowCol.col}
+    <div
+      className="container"
+      onClick={(evnt) => {
+        // debug this using this page: https://www.reddit.com/r/typescript/comments/v5hzws/property_classname_does_not_exist_on_type/
+        const target = evnt.target as HTMLTextAreaElement;
+        const element = target.className;
+        if (element === "container") {
+          setShowCarousel(false);
+        }
+      }}
     >
-      {images.map((img, index) => (
-        <ImageWrap>
-          <ImageItem src={img} />
-        </ImageWrap>
-      ))}
-    </ImageGrid>
+      {showCarousel ? (
+        <>
+          <div
+            style={{
+              height: "90vh",
+              maxWidth: "90%",
+              margin: "0 auto",
+              cursor: "pointer",
+            }}
+          >
+            <Carousel imgList={images} selectedImgIndex={selectedImgIndex} />
+          </div>
+        </>
+      ) : (
+        <ImageGrid
+          height={imagesGridHeight}
+          maxWidth={imagesGridMaxWidth}
+          numberOfImgs={numberOfImgs}
+          row={rowCol.row}
+          col={rowCol.col}
+        >
+          {numberOfImgs <= 5 &&
+            images
+              .slice(0, numberOfImgs)
+              .map((photo: UnsplashPhotoFortmat, index) => (
+                <ImageWrap key={index} onClick={useCarousel.bind(this, index)}>
```

Comment:
whyyy do you use bind here and below? 
```suggestion
                <ImageWrap key={index} onClick={() => useCarousel(index)}>
```

Reply:
this place I use bind bcs I want when user click to the image, we can pass the index of that image into the useCarousel parameter.

Reply:
what's the difference between that and my suggestions?:)

Reply:
In general, this is not a mistake (so it's not incorrect) but no one would normally use it with latest React and modern JS

Reply:
oh yah, I never see your suggestion. Sorry, I am still quite new on PR comment, so I mis-understand abit

Reply:
I have just try your suggestion. I work also. thank you so much. :)

Reply:
actually earlier, I follow your suggestion, then it showed the error: __"Line 130:55:  React Hook "useCarousel" cannot be called inside a callback. React Hooks must be called in a React function component or a custom React Hook function  react-hooks/rules-of-hooks"_
Then I try using "bind", It works with the function name "useCarousel" ... 
but then I understand that the error happen is Because the function name raise the issue haha =))) 
Thank you to highlight the function name also

Reply:
yep correct `use-` prefix is reserved for Hooks in react and might throw this kind of error